hw/ip/otbn/dv/rig/rig/gens/edge_load_store.py

```diff
@@ -0,0 +1,234 @@
+                               'generator.')
+
+        self.grs1_op_type = self.sw.operands[0].op_type
+
+        self.imm_op_type = self.sw.operands[1].op_type
+        assert self.imm_op_type.shift == 0
+
+        self.grs2_op_type = self.sw.operands[2].op_type
+
+        self.grd_op_type = self.lw.operands[0].op_type
+        # BN.LID Instruction Checks
+
+        self.bnlid = self._get_named_insn(insns_file, 'bn.lid')
+
+        self.bn_imm_op_type = self.bnlid.operands[2].op_type
+        # bn.lid expects the operands: grd, grs1, offset, grs1_inc, grd_inc
+        if len(self.bnlid.operands) != 5:
+            raise RuntimeError('Unexpected number of operands for bn.lid')
+
+        if not (isinstance(self.bnlid.operands[0].op_type, RegOperandType) and
+                (self.bnlid.operands[0].op_type.reg_type == 'gpr' and
+                not self.bnlid.operands[0].op_type.is_dest() and
+                isinstance(self.bnlid.operands[1].op_type, RegOperandType) and
+                self.bnlid.operands[1].op_type.reg_type == 'gpr' and
+                not self.bnlid.operands[1].op_type.is_dest() and
+                isinstance(self.bnlid.operands[2].op_type, ImmOperandType) and
+                self.bnlid.operands[2].op_type.signed and
+                isinstance(self.bnlid.operands[3].op_type, OptionOperandType) and
+                isinstance(self.bnlid.operands[4].op_type, OptionOperandType))):
+            raise RuntimeError('BN.LID instruction from instructions file is'
+                               'not the shape expected by MisalignedBadLoadStore'
+                               'generator.')
+
+        # BN.SID Instruction Checks
+
+        self.bnsid = self._get_named_insn(insns_file, 'bn.sid')
+
+        # bn.sid expects the operands: grs1, grs2, offset, grs1_inc,
+        # grs2_inc
+        if len(self.bnsid.operands) != 5:
+            raise RuntimeError('Unexpected number of operands for bn.sid')
+
+        if not (isinstance(self.bnsid.operands[0].op_type, RegOperandType) and
+                self.bnsid.operands[0].op_type.reg_type == 'gpr' and
+                not self.bnsid.operands[0].op_type.is_dest() and
+                isinstance(self.bnsid.operands[1].op_type, RegOperandType) and
+                self.bnsid.operands[1].op_type.reg_type == 'gpr' and
+                not self.bnsid.operands[1].op_type.is_dest() and
+                isinstance(self.bnsid.operands[2].op_type, ImmOperandType) and
+                self.bnsid.operands[2].op_type.signed and
+                isinstance(self.bnsid.operands[3].op_type, OptionOperandType) and
+                isinstance(self.bnsid.operands[4].op_type, OptionOperandType)):
+            raise RuntimeError('BN.SID instruction from instructions file is'
+                               'not the shape expected by MisalignedBadLoadStore'
+                               'generator.')
+
+        for insn in insns_file.insns:
```

Comment:
Probably iterate over `[self.sw, self.lw, self.bnlid, self.bnsid]` and drop the first "if"?

Also, I don't think you actually need to store the instructions by name in the class instance above: maybe just stick them in local variables and add them to `self.insns` here?